coba/learners/vowpal.py

```diff
@@ -196,45 +196,39 @@ class VowpalLearner(Learner):
 
     @staticmethod
     def make_args(
-        options: Sequence[str],
-        noconstant: bool,
-        interactions: Sequence[str],
-        ignore_linear:Sequence[str],
-        seed: Optional[int],
-        **kwargs) -> str:
-        """Turn settings into a VW command line string.
+            vw_kwargs: Dict[str, Any],
+            namespace_interactions: Sequence[str] = DEFAULT_NAMESPACE_INTERACTIONS,
+    ) -> Sequence[str]:
+        """
+        Turn settings into VW command line arguments.
 
-        Args:
-            options: A sequence of string values that represent VW CLI options.
-            noconstant: Indicates if constant term should be included in the VW features.
-            interactions: A sequence of namespace interactions to use during learning.
-            ignore_linear: A sequence of linear namespaces to ignore during learning.
-            seed: A random number generator seed to make sure VW behaves consistently.
-            kwargs: Any number of additional options to add to the arg string.
+         Args:
+            vw_kwargs: Keyword argument dict that gets translated into VW CLI arguments
+            namespace_interactions: A sequence of namespace interactions to use during learning.
         """
+        args = []
+        vw_kwargs['quiet'] = vw_kwargs.get('quiet', True)
+
+        for k,v in vw_kwargs.items():
+            if v is not False and v is not None:
+                if not k.startswith("-"):
+                    k = ("-" if len(k)==1 else "--") + k
+                args.append(k if (v is True) else f"{k} {v}")
 
-        options = list(filter(None,options))
+        noconstant = sum([n for n in namespace_interactions if isinstance(n, (int, float))]) == 0
+        ignore_linear = set(['x', 'a']) - set(namespace_interactions)
+        interactions = [n for n in namespace_interactions if isinstance(n, str) and len(n) > 1]
 
         if noconstant:
-            options.append(f"--noconstant")
+            args.append(f"--noconstant")
 
         for interaction in interactions:
-            options.append(f"--interactions {interaction}")
+            args.append(f"--interactions {interaction}")
 
         for ignore in ignore_linear:
-            options.append(f"--ignore_linear {ignore}")
-
-        if seed is not None:
-            options.append(f"--random_seed {seed}")
+            args.append(f"--ignore_linear {ignore}")
 
-        kwargs['quiet'] = kwargs.get('quiet',True)
-
-        for k,v in kwargs.items():
-            if v is not False:
-                k = ("-" if len(k)==1 else "--") + k
-                options.append(k if (v is None or v is True) else f"{k} {v}")
-
-        return " ".join(options)
+        return args
```

Comment:
Return args as collection for testability and potential future constructors that don't depend on an assembled config string.